docs/sql-programming-guide.md

```diff
@@ -12,130 +12,130 @@ title: Spark SQL and DataFrames
 Spark SQL is a Spark module for structured data processing. Unlike the basic Spark RDD API, the interfaces provided
 by Spark SQL provide Spark with more information about the structure of both the data and the computation being performed. Internally,
 Spark SQL uses this extra information to perform extra optimizations. There are several ways to
-interact with Spark SQL including SQL, the DataFrames API and the Datasets API. When computing a result
+interact with Spark SQL including SQL and the Datasets API. When computing a result
 the same execution engine is used, independent of which API/language you are using to express the
-computation. This unification means that developers can easily switch back and forth between the
-various APIs based on which provides the most natural way to express a given transformation.
+computation. This unification means that developers can easily switch back and forth between
+different APIs based on which provides the most natural way to express a given transformation.
 
 All of the examples on this page use sample data included in the Spark distribution and can be run in
 the `spark-shell`, `pyspark` shell, or `sparkR` shell.
 
 ## SQL
 
-One use of Spark SQL is to execute SQL queries written using either a basic SQL syntax or HiveQL.
```

Comment:
The "basic SQL" used to refer to the simple SQL dialect we supported in 1.x but removed in 2.0.